torch/csrc/distributed/c10d/ProcessGroupUCC.cpp

```diff
@@ -128,11 +129,58 @@ std::map<std::string, std::string> torch_ucc_envs_map = {
     {"TORCH_UCC_ENABLE_COMMS_LOGGER", "0"},
 };
 
+// trim implementation borrowed from https://stackoverflow.com/a/17976541
```

Comment:
Would be nice to add a comment here stating what this function is doing.

Reply:
added